src/VehicleSetup/JoystickConfigButtons.qml

```diff
@@ -163,24 +165,85 @@ ColumnLayout {
+                                // If there is not firmware function, check QGC ones
+                                var i = find(_activeJoystick.buttonActions[modelData])
+                                currentIndex = i
+                            }
+                        }
+                    }
+
+                    Component.onCompleted:  _findCurrentButtonAction()
+                    onModelChanged:         _findCurrentButtonAction()
+                    onActivated:            function (optionIndex) {
+                        var func = textAt(optionIndex)
+                        if (factOptions.indexOf(func) > -1) {
+                            // This is a FW action, set parameter to the action and set QGC's handler to No Action
+                            fact.value = factOptions.indexOf(func)
+                            _activeJoystick.setButtonAction(modelData, "No Action")
+                        } else {
+                            // This is a QGC action, set parameters to Disabled and QGC to the desired action
+                            _activeJoystick.setButtonAction(modelData, func)
+                            fact.value = 0
+                            fact_shift.value = 0
+                        }
+                    }
+                }
+                QGCCheckBox {
+                    id:                         repeatCheck
+                    text:                       qsTr("Repeat")
+                    enabled:                    currentAssignableAction && _activeJoystick.calibrated && currentAssignableAction.canRepeat
+                    visible:                    !globals.activeVehicle.supportsJSButton
+
+                    onClicked: {
+                        _activeJoystick.setButtonRepeat(modelData, checked)
+                    }
+                    Component.onCompleted: {
+                        if(_activeJoystick) {
+                            checked = _activeJoystick.getButtonRepeat(modelData)
+                        }
+                    }
+                    anchors.verticalCenter:     parent.verticalCenter
+                }
+                Item {
+                    width:                      ScreenTools.defaultFontPixelWidth * 2
+                    height:                     1
                 }
 
                 FactComboBox {
-                    id:             shiftJSButtonActionCombo
-                    width:          ScreenTools.defaultFontPixelWidth * 15
-                    fact:           hasFirmwareSupport ? controller.getParameterFact(-1, parameterShiftName) : null;
-                    visible:        hasFirmwareSupport
-                    indexModel:     false
+                    id:         shiftJSButtonActionCombo
+                    width:      ScreenTools.defaultFontPixelWidth * 26
+                    fact:       controller.parameterExists(-1, `BTN${index}_SFUNCTION`) ? controller.getParameterFact(-1, `BTN${index}_SFUNCTION`) : null;
```

Comment:
Should we make a helper function to avoid maintaining this string around the code ?
The same thing for `BTN${INDEX}_FUNCTION`